src/main.rs

```diff
@@ -1,126 +1,267 @@
-use std::fs::File;
-use std::io::Error as IOError;
+mod cel;
+mod light;
+mod outline;
+mod color;
+mod scale;
+mod geometry;
+mod material;
 
-use fbx_direct::reader::{EventReader, FbxEvent::*, Error as FBXError};
-use nalgebra::Point2D; //TODO: Add the nalgebra crate and lookup the closest struct to Point2D
+use std::f32::consts::PI;
+use std::time::Duration;
+use std::path::Path;
+use std::thread;
+use std::rc::Rc;
 
-// TODO: Pick a better name than AppError
-#[derive(Debug)]
-enum AppError {
-    IOError(IOError),
-    FBXError(FBXError),
+use euc::{Pipeline, rasterizer, buffer::Buffer2d, Target};
+use minifb::{self, Key, KeyRepeat};
+use tobj;
+use vek::{Mat4, Vec2, Vec3, Vec4, Rgba};
+use image::ImageBuffer;
+
+use crate::cel::CelShader;
+use crate::outline::OutlineShader;
+use crate::light::DiffuseLight;
+use crate::scale::scale_buffer;
+use crate::geometry::Mesh;
+use crate::material::Material;
+use crate::color::{rgba_to_bgra_u32, bgra_u32_to_rgba, vek_rgba_to_image_rgba};
+
+fn main() {
+    let image_width = 64;
+    let image_height = 64;
+
+    let frames: Vec<_> = (1..=8).map(|i| {
+        load_frame(&format!("samples/bigboi/obj/bigboi_rigged_{:06}.obj", i))
+    }).collect();
+
+    // The transformation that represents the center of the model, all points in the model are
+    // relative to this
+    // Also known as the "world" transformation
+    //
+    // Model coordinates -> World coordinates
+    let model = Mat4::identity();
+    // The transformation that represents the position and orientation of the camera
+    //
+    // World coordinates -> Camera coordinates
+    let view = Mat4::rotation_x(PI/8.0) * Mat4::rotation_y(0.0*PI/2.0);
```

Comment:
[MINOR] later on we'll probably want to wrap all of these matrices in a Camera type :)